internal/core/software_renderer.rs

```diff
@@ -932,7 +935,12 @@ impl<'a, T: ProcessScene> SceneBuilder<'a, T> {
                             + source_rect.origin.y as usize
                             - t.rect.origin.y as usize;
                         let stride = t.rect.width() as u16 * t.format.bpp() as u16;
-                        let t_alpha_u16 = t.color.alpha() as u16;
+                        let color = if colorize.alpha() > 0 { colorize } else { t.color };
+                        let alpha = if colorize.alpha() > 0 || t.format == PixelFormat::AlphaMap {
+                            color.alpha() as u16 * alpha_u16 / 255
+                        } else {
+                            alpha_u16
```

Comment:
I suggest to rename `alpha_u16` to `global_alpha_u16` and `RenderState`'s `fn alpha_u8` to `global_alpha_u8` for what I think would be improved readability.